### PR TITLE
nsz: init at 4.1.0

### PR DESCRIPTION
**Pull request by eyJhb**
###### Description of changes

Added nsz to nixpkgs. 
doCheck is false, as the library requires production keys to be dumped from a Nintendo Switch.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/nsz/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildPythonPackage, fetchFromGitHub, pycryptodome, enlighten, zstandard, kivy
+, withGUI ? true
```
Comment (markuskowa) [inline pkgs/development/python-modules/nsz/default.nix:2]:
```suggestion
, withGUI ? true
, kivy
```
(the optional dependency should be grouped with the switch)

File: pkgs/development/python-modules/nsz/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  version = "4.1.0";
+
+  src = fetchFromGitHub rec {
+    owner = "nicoboss";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-tdngXV+VUOAkg3lF2NOmw0mBeSEE+YpUfuKukTKcPnM=";
+  };
+
+  # doCheck = false;
```
Comment (markuskowa) [inline pkgs/development/python-modules/nsz/default.nix:16]:
```suggestion
```

File: pkgs/development/python-modules/nsz/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildPythonPackage, fetchFromGitHub, pycryptodome, enlighten, zstandard, kivy
+, withGUI ? true
+}:
+
+buildPythonPackage rec {
+  pname = "nsz";
+  version = "4.1.0";
+
+  src = fetchFromGitHub rec {
```
Comment (markuskowa) [inline pkgs/development/python-modules/nsz/default.nix:9]:
```suggestion
  src = fetchFromGitHub {
```

**Review by markuskowa — CHANGES_REQUESTED** [review]

**Comment by eyJhb** [comment]
@markuskowa  everything should be resolved now :) If you could please double check and approve to make sure, it be highly appreciated!

Thanks for reviewing!

File: pkgs/development/python-modules/nsz/default.nix (hunk 2 above)
Comment (markuskowa) [reply pkgs/development/python-modules/nsz/default.nix:17]: That commented out `# doCheck = false;` should be removed.

Comment (eyJhb) [reply pkgs/development/python-modules/nsz/default.nix]: I'm impressed that I saw it, and still missed it. Should be done now.

**Review by markuskowa — APPROVED** [review]

**Pull request merged**